### PR TITLE
fix(node): always use .RunCommand when binary is named `node`

### LINKED ISSUE #7610: bun vs node cli prevents some lifecycle scripts from working
### What version of Bun is running?

_No response_

### What platform is your computer?

_No response_

### What steps can reproduce the bug?

Related to #7605 

```
mkdir scripts
touch scripts/hello.js
```

```
bun --bun run node scripts/hello
```

### What is the expected behavior?

it should resolve and run the script since node can

### What do you see instead?

error: script not found "scripts/hello"

### Additional information

cc @dylan-conway

### PR DESCRIPTION
**Pull request by paperclover**
### What does this PR do?

This makes `bun --bun node` always use the run command. Fixes #7610.

image

Related issue #7605, but this just fixes the underlying case that prompted wanted having a feature to disable a single script. I think its fine to keep that open as there are legitimate reasons to disallow a single script, and that probably can come with a more robust `trustedDependencies` field.

Also, this tweaks some error messages to be more consistent with each other.

image

also addresses #7504 by using `/tmp/bun-node-<git sha>` however this is not fully closing the issue as it does not actually fix the issue it describes, but simply makes it harder to hit.

### How did you verify your code works?

I wrote tests

### CODE REVIEW

File: src/cli/run_command.zig [hunk 1]
```diff
@@ -17,7 +17,8 @@ const WaiterThread = JSC.Subprocess.WaiterThread;
 const lex = bun.js_lexer;
 const logger = @import("root").bun.logger;
 const clap = @import("root").bun.clap;
-const Arguments = @import("../cli.zig").Arguments;
+const cli = @import("../cli.zig");
```
Comment (Jarred-Sumner) [inline src/cli/run_command.zig]:
```suggestion
const cli = bun.CLI;
```

File: src/cli/run_command.zig [hunk 2]
```diff
@@ -1731,6 +1732,43 @@ pub const RunCommand = struct {
 
         return false;
     }
+
+    pub fn execAsIfNode(ctx: Command.Context) !void {
+        std.debug.assert(cli.pretend_to_be_node);
+
+        if (ctx.positionals.len == 0) {
```
Comment (Jarred-Sumner) [inline src/cli/run_command.zig:1757]: we could technically make this open `bunx bun-repl`

File: src/cli/run_command.zig [hunk 3]
```diff
@@ -1731,6 +1732,43 @@ pub const RunCommand = struct {
...
+        // var win_resolver = resolve_path.PosixToWinNormalizer{};
+
+        const filename = ctx.positionals[0];
+
+        const normalized_filename = if (std.fs.path.isAbsolute(filename))
+            // TODO(@paperdave): merge windows branch
+            // try win_resolver.resolveCWD("/dev/bun/test/etc.js");
+            filename
+        else brk: {
+            const cwd = try bun.getcwd(&path_buf);
```
Comment (Jarred-Sumner) [inline src/cli/run_command.zig:1772]: we haven't already loaded the cwd at this point?

Comment (paperclover) [reply src/cli/run_command.zig:1772]: i dont think so. code copied lightly from the existing code path

**Pull request merged**